[WIP]: IBMZ updates for HCP 4.17 (#83206)
openshift/openshift-docs: modules/hcp-ibmz-adding-reg-ca-hostedcluster.adoc

:_mod-docs-content-type: CONCEPT
[id="hcp-ibmz-adding-registry-ca-hostedcluster_{context}"]

> = Adding the Registry Certificate Authority to the Hosted Cluster

Reviewer: ```suggestion
= Adding the registry certificate authority to the hosted cluster
```

> When you are creating {hcp}, include the `additional-trust-bundle` and `image-content-sources`. This will allow the hosted cluster to inject the Certificate Authority into the Data plane workers so that the images are pulled from the registry.

Reviewer: ```suggestion
When you are deploying {hcp} on {ibm-z-title} in a disconnected environment, include the `additional-trust-bundle` and `image-content-sources` resources. Those resources allow the hosted cluster to inject the certificate authority into the data plane workers so that the images are pulled from the registry.
```
Check my revision for accuracy. I assumed that `additional-trust-bundle` and `image-content-sources` are resources.

> . Create `icsp.yaml` with image-content-sources. The `image-content-sources` information is available in the `ImageContentSourcePolicy` generated after mirrorring the images using `oc-mirror`.

Reviewer: ```suggestion
. Create the `icsp.yaml` file with the `image-content-sources` information. The `image-content-sources` information is available in the `ImageContentSourcePolicy` YAML file that is generated after you mirror the images by using `oc-mirror`.
```
Check my revision for accuracy.

+
.Example ImageContentSourcePolicy file
[source,terminal]
----
# cat icsp.yaml 
- mirrors:
  - <mirror_registry>/openshift/release
  source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
- mirrors:
  - <mirror_registry>/openshift/release-images
  source: quay.io/openshift-release-dev/ocp-release
----

> . Provide the `additional-trust-bundle` certificate during the {hcp} creation to update the compute nodes with the certificates. Below is the sample {hcp} command to create the compute nodes with the certificates.

Reviewer: Add a blank line between lines 19 and 20.

> . Provide the `additional-trust-bundle` certificate during the {hcp} creation to update the compute nodes with the certificates. Below is the sample {hcp} command to create the compute nodes with the certificates.

Reviewer: ```suggestion
. When you create a hosted cluster, provide the `additional-trust-bundle` certificate to update the compute nodes with the certificates:
```

[source,terminal]
----
# cat icsp.yaml 
- mirrors:
  - <mirror_registry>/openshift/release
  source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
- mirrors:
  - <mirror_registry>/openshift/release-images
  source: quay.io/openshift-release-dev/ocp-release
----

. When you create a hosted cluster, provide the `additional-trust-bundle` certificate to update the compute nodes with the certificates:
+
[source,terminal]
----
$ hcp create cluster agent \
    --name=<hosted_cluster_name> \ // <1>
    --pull-secret=<path_to_pull_secret> \ // <2>
    --agent-namespace=<hosted_control_plane_namespace> \ // <3>
    --base-domain=<basedomain> \ // <4>
    --api-server-address=api.<hosted_cluster_name>.<basedomain> \
    --etcd-storage-class=<etcd_storage_class> \ // <5>
    --ssh-key  <path_to_ssh_public_key> \ // <6>
    --namespace <hosted_cluster_namespace> \ // <7>
    --control-plane-availability-policy SingleReplica \
    --release-image=quay.io/openshift-release-dev/ocp-release:<ocp_release_image> \ // <7>
    --additional-trust-bundle <path for cert> \ // <8>
    --image-content-sources icsp.yaml
----
<1> Replace `<hosted_cluster_name>` with the name of your hosted cluster.

> <2> Replace the path to your pull secret, for example, `/user/name/pullsecret`.

Author: Checked with the team, retained as it is a good-to-have info

<3> Replace `<hosted_control_plane_namespace>` with the name of the hosted control plane namespace, for example, `clusters-hosted`.
<4> Replace the name with your base domain, for example, `example.com`.
<5> Replace the etcd storage class name, for example, `lvm-storageclass`.
<6> Replace the path to your SSH public key. The default file path is `~/.ssh/id_rsa.pub`.

> <7> Replace with the supported {ocp-short} version that you want to use, for example, `4.17.0-multi`.

Reviewer: ```suggestion
<7> Replace with the supported {product-title} version that you want to use, for example, `4.17.0-multi`.
```

    --pull-secret=<path_to_pull_secret> \ // <2>
    --agent-namespace=<hosted_control_plane_namespace> \ // <3>
    --base-domain=<basedomain> \ // <4>
    --api-server-address=api.<hosted_cluster_name>.<basedomain> \
    --etcd-storage-class=<etcd_storage_class> \ // <5>
    --ssh-key  <path_to_ssh_public_key> \ // <6>
    --namespace <hosted_cluster_namespace> \ // <7>
    --control-plane-availability-policy SingleReplica \
    --release-image=quay.io/openshift-release-dev/ocp-release:<ocp_release_image> \ // <8>
    --<additional-trust-bundle> /var/lib/libvirt/registry/certs/registry.crt \ // <9>
    --image-content-sources icsp.yaml
----
<1> Replace `<hosted_cluster_name>` with the name of your hosted cluster.
<2> Replace the path to your pull secret, for example, `/user/name/pullsecret`.
<3> Replace `<hosted_control_plane_namespace>` with the name of the hosted control plane namespace, for example, `clusters-hosted`.
<4> Replace the name with your base domain, for example, `example.com`.
<5> Replace the etcd storage class name, for example, `lvm-storageclass`.
<6> Replace the path to your SSH public key. The default file path is `~/.ssh/id_rsa.pub`.

> <7> Replace `<hosted_cluster_namespace>` with your hosted cluster. Replace `<hosted_cluster_namespace>` with the name of your hosted cluster namespace.

Reviewer 2: <hosted_cluster_namespace> 
here we are referring  <hosted_cluster_namespace> 2 times. 
we can give like this for <7>
Replace `<hosted_cluster_namespace>` with the name of your hosted cluster namespace.

Author: You are right, there was a repetition, i have removed <3> foot note.

<8> Replace with the supported {ocp-short} version that you want to use, for example, `4.17.0-multi`.

> <9> The `additional-trust-bundle-file`, `http-proxy`, and `https-proxy` arguments are optional.

Reviewer 2: additional-trust-bundle-file  argument is not optional , as we are in the context of disconnected installation.It is a mandatory argument.

and any specific reason for giving `http-proxy`, and `https-proxy` , even though we are not giving in command.?

Reviewer 2: but here the path shoulbe the variable 
--additional-trust-bundle <path for cert>
